driver/terraform.go

```diff
@@ -224,24 +225,78 @@ func (tf *Terraform) ApplyTask(ctx context.Context) error {
 }
 
 // UpdateTask updates the task on the driver. Makes any calls to re-init
-// depending on the fields updated.
-func (tf *Terraform) UpdateTask(patch PatchTask) error {
+// depending on the fields updated. If update task is requested with the inspect
+// run option, then returns the inspected plan
+func (tf *Terraform) UpdateTask(ctx context.Context, patch PatchTask) (string, error) {
+	switch patch.RunOption {
+	case "", RunOptionInspect, RunOptionNow:
+		// valid options
+	default:
+		return "", fmt.Errorf("Invalid run option '%s'. Please select a valid "+
+			"option", patch.RunOption)
+	}
+
 	tf.mu.Lock()
 	defer tf.mu.Unlock()
 
 	reinit := false
 
 	if tf.task.Enabled != patch.Enabled {
 		tf.task.Enabled = patch.Enabled
-		reinit = true
+		if tf.task.Enabled {
+			reinit = true
+		}
+	}
+
+	// identify cases where resources are not impacted and we can return early
+	switch {
+	case patch.Enabled == false:
+		return "", nil
+	default:
+		// onward!
 	}
 
 	if reinit {
+		if err := tf.initTask(true); err != nil {
+			return "", fmt.Errorf("Error updating task '%s'. Unable to init "+
+				"task: %s", tf.task.Name, err)
+		}
 		tf.inited = false
-		return tf.initTask(true)
+
+		for i := int64(0); ; i++ {
+			rendered, err := tf.renderTemplate(ctx)
+			if err != nil {
+				return "", fmt.Errorf("Error updating task '%s'. Unable to "+
+					"render template for task: %s", tf.task.Name, err)
+			}
+			if rendered {
+				break
+			}
+			tf.logDepSize(50, i)
+		}
 	}
 
-	return nil
+	if patch.RunOption == RunOptionInspect {
+		log.Printf("[TRACE] (driver.terraform) update task '%s'. inspect run "+
+			"option", tf.task.Name)
+		plan, err := tf.inspectTask(ctx, true)
+		if err != nil {
+			return "", fmt.Errorf("Error updating task '%s'. Unable to inspect "+
+				"task: %s", tf.task.Name, err)
+		}
+
+		plan = strings.ReplaceAll(plan, "Terraform", "CTS")
```

Comment:
Can you expand the benefits of modifying the Terraform plan output?

Reply:
Yea, so this plan will be outputted in the enable CLI something like:
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

CTS will perform the following actions:

 // diff
```

It looked a little confusing to me to say "Terraform will perform the following actions" but could just be me. 🤔 it felt a bit wrong in spirit to replace "Terraform" though too. Maybe `strings.ReplaceAll(plan, "Terraform", "Terraform (on behalf of CTS)")`? Do you have thoughts?

Reply:
I see thanks for the example output!

> it felt a bit wrong in spirit to replace "Terraform"

Yeah I share that sentiment too, but I can see the value you're bringing to bridge that CTS is calling Terraform. How about appending the CTS context before Terraform's output?

```
<CTS info here executing Terraform plan>

An execution plan has been generated...
  + create

Terraform will perform the following actions:

  // diff
```

Reply:
I like the idea of a preface mentioning Terraform! There is actually a preface in the CLI-side prior to printing out the plan. I might actually modify the preface in the CLI to mention Terraform there instead and keep this just the 'pure' plan. I'll delete the string parsing here. Thanks!